### PR TITLE
Fix for issue #627

### PR DESCRIPTION
**Pull request by madhuthorat**
Signed-off-by: Madhu Thorat <madhu.punjabi@in.ibm.com>

## Pull request checklist

```release-notes

```

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?
Snapshots failing for remote cluster with different file names

## What is the new behavior?
Fix for - snapshots failing for remote cluster with different file names

## How risky is this change?
- [ ] Small, isolated change
- [x] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

File: driver/csiplugin/controllerserver.go [hunk 1]
```diff
@@ -1831,7 +1836,13 @@ func (cs *ScaleControllerServer) CreateSnapshot(ctx context.Context, req *csi.Cr
...
 	}
 
-	filesystemName, err := conn.GetFilesystemName(volumeIDMembers.FsUUID)
+	primaryConn, isprimaryConnPresent := cs.Driver.connmap["primary"]
+	if !isprimaryConnPresent {
+		glog.Errorf("CreateSnapshot - unable to get connector for primary cluster")
+		return nil, status.Error(codes.Internal, "CreateSnapshot - unable to find primary cluster details in custom resource")
+	}
+
+	filesystemName, err := primaryConn.GetFilesystemName(volumeIDMembers.FsUUID)
```
[the remark this answers is not in the document]
Comment (madhuthorat) [reply driver/csiplugin/controllerserver.go:1845]: This change is similar to code in DeleteVolume which receives volume ID. If DeleteVolume works well, this should work too !

[the remark this answers is not in the document]
Comment (madhuthorat) [reply driver/csiplugin/controllerserver.go:1845]: This change is similar to code in DeleteVolume which receives volume ID. If DeleteVolume works well, this should work too !

File: driver/csiplugin/controllerserver.go [hunk 2]
```diff
@@ -941,7 +941,12 @@ func (cs *ScaleControllerServer) copySnapContent(scVol *scaleVolume, snapId scal
 		return err
 	}
 
-	targetFsDetails, err := conn.GetFilesystemDetails(targetFsName)
+	primaryConn, isprimaryConnPresent := cs.Driver.connmap["primary"]
+	if !isprimaryConnPresent {
+		glog.Errorf("unable to get connector for primary cluster")
+		return status.Error(codes.Internal, "unable to find primary cluster details in custom resource")
+	}
+	targetFsDetails, err := primaryConn.GetFilesystemDetails(targetFsName)
```
Comment (deeghuge) [inline driver/csiplugin/controllerserver.go:949]: primary cluster won't be aware of targetFsName as it is name of filesystem on remote cluster

Comment (madhuthorat) [reply driver/csiplugin/controllerserver.go]: Removed the change.

**Review by deeghuge — APPROVED** [review]

**Pull request merged**